src/util/plugin.js

```diff
@@ -0,0 +1,37 @@
+const assert = require('assert');
+
+const pluginTypes = ['FILTER', 'INJECT', 'SORT'];
+
+const join = (input) => {
+  const result = input.filter(e => !!e).join('.');
+  if (result === '*') {
+    return '';
+  }
+  if (result.endsWith('.*')) {
+    return result.slice(0, -1);
+  }
+  return result;
+};
+
+const plugin = (type, options) => {
+  assert(pluginTypes.includes(type));
+  assert(options instanceof Object && !Array.isArray(options));
+  assert(Object.keys(options).length === 3);
```

Comment:
Joi validation?

Reply:
```js
Joi.object().keys({
  target: Joi.string(),
  requires: Joi.items(Joi.string()),
  fn: Joi.func()
})
```

Reply:
Joi is no longer a dependency of this package. So no

Reply:
joi is not a dependency, not really wanting to make it one